pyquil/tests/test_paulis.py

```diff
@@ -185,40 +191,47 @@ def test_pauli_sum():
     q_plus = 0.5 * PauliTerm('X', 0) + 0.5j * PauliTerm('Y', 0)
     the_sum = q_plus * PauliSum([PauliTerm('X', 0)])
     term_strings = [str(x) for x in the_sum.terms]
-    assert '0.5*I' in term_strings
+    assert '(0.5+0j)*I' in term_strings
     assert '(0.5+0j)*Z0' in term_strings
     assert len(term_strings) == 2
     assert len(the_sum.terms) == 2
 
     the_sum = q_plus * PauliTerm('X', 0)
     term_strings = [str(x) for x in the_sum.terms]
-    assert '0.5*I' in term_strings
+    assert '(0.5+0j)*I' in term_strings
     assert '(0.5+0j)*Z0' in term_strings
     assert len(term_strings) == 2
     assert len(the_sum.terms) == 2
 
     the_sum = PauliTerm('X', 0) * q_plus
     term_strings = [str(x) for x in the_sum.terms]
-    assert '0.5*I' in term_strings
+    assert '(0.5+0j)*I' in term_strings
     assert '(-0.5+0j)*Z0' in term_strings
     assert len(term_strings) == 2
     assert len(the_sum.terms) == 2
 
+    with pytest.raises(ValueError):
+        _ = PauliSum(sI(0))
+    with pytest.raises(ValueError):
+        _ = PauliSum([1, 1, 1, 1])
+    with pytest.raises(ValueError):
+        _ = the_sum * []
+
 
 def test_ps_adds_pt_1():
     term = ID
     b = term + term
-    assert str(b) == "2.0*I"
-    assert str(b + term) == "3.0*I"
-    assert str(term + b) == "3.0*I"
+    assert str(b) == "(2+0j)*I"
+    assert str(b + term) == "(3+0j)*I"
+    assert str(term + b) == "(3+0j)*I"
 
 
 def test_ps_adds_pt_2():
     term = ID
     b = term + 1.0
-    assert str(b) == "2.0*I"
-    assert str(b + 1.0) == "3.0*I"
-    assert str(1.0 + b) == "3.0*I"
+    assert str(b) == "(2+0j)*I"
+    assert str(b + 1.0) == "(3+0j)*I"
+    assert str(1.0 + b) == "(3+0j)*I"
```

Comment:
missing test cases are if you add `ID` and `X` or any other kind of different paulis